### PR TITLE
refactor: Models feature parity

### PR DESCRIPTION
**Pull request by KevLehman**
## Further comments

This PR migrates all model functions from meteor models under server/models to raw models.
EE models will be done separately, as they require other things before.

ARCH-172

### CODE REVIEW

File: apps/meteor/app/importer/server/classes/ImportDataConverter.ts [hunk 1]
```diff
@@ -371,6 +371,8 @@ export class ImportDataConverter {
 				this.saveError(_id, e instanceof Error ? e : new Error(String(e)));
 			}
 		});
+
+		await Promise.all(promises);
```
Comment (murtaza98) [inline apps/meteor/app/importer/server/classes/ImportDataConverter.ts:375]: Wondering if we should use AllSettled here instead. So if any one of the users has some record that isn't valid and is throwing errors then only that particular user won't be imported. The rest of the users will still get imported this way.

File: apps/meteor/ee/server/models/raw/LivechatRooms.ts [hunk 2]
```diff
@@ -32,6 +32,7 @@ declare module '@rocket.chat/model-typings' {
 	}
 }
 
+// @ts-expect-error - Model is in JS, and types are getting weird :)
```
Comment (murtaza98) [inline apps/meteor/ee/server/models/raw/LivechatRooms.ts:35]: 😢

File: apps/meteor/server/models/raw/LivechatRooms.js [hunk 3]
```diff
@@ -12,6 +13,40 @@ export class LivechatRoomsRaw extends BaseRaw {
 		super(db, 'room', trash);
 	}
 
+	// move indexes from constructor to here using IndexDescription as type
```
Comment (murtaza98) [inline apps/meteor/server/models/raw/LivechatRooms.js:16]:
```suggestion
```

File: apps/meteor/server/models/raw/LivechatDepartmentAgents.ts [hunk 4]
```diff
@@ -113,4 +150,208 @@ export class LivechatDepartmentAgentsRaw extends BaseRaw<ILivechatDepartmentAgen
...
+		return this.updateOne(
+			{
+				agentId: agent.agentId,
+				departmentId: agent.departmentId,
+			},
+			{
+				$set: {
+					username: agent.username,
+					departmentEnabled: agent.departmentEnabled,
+					count: parseInt(`${agent.count}`),
```
Comment (murtaza98) [inline apps/meteor/server/models/raw/LivechatDepartmentAgents.ts:179]:
```suggestion
					count: parseInt(`${agent.count}`),
					order: parseInt(`${agent.order}`),
```
Aren't they already numbers now?

File: apps/meteor/server/models/raw/LivechatRooms.js [hunk 5]
```diff
@@ -12,6 +13,40 @@ export class LivechatRoomsRaw extends BaseRaw {
...
+	modelIndexes() {
```
Comment (murtaza98) [inline apps/meteor/server/models/raw/LivechatRooms.js:17]:
```suggestion
	protected modelIndexes() {
```

File: apps/meteor/server/models/raw/LivechatDepartment.ts [hunk 6]
```diff
@@ -151,4 +188,115 @@ export class LivechatDepartmentRaw extends BaseRaw<ILivechatDepartmentRecord> im
...
+
+	findEnabledWithAgents(projection: FindOptions<ILivechatDepartment>['projection'] = {}): FindCursor<ILivechatDepartment> {
+		const query = {
+			numAgents: { $gt: 0 },
+			enabled: true,
+		};
+		return this.find(query, projection && { projection });
+	}
+
+	findEnabledWithAgentsAndBusinessUnit(
```
Comment (murtaza98) [inline apps/meteor/server/models/raw/LivechatDepartment.ts:249]: I see an override for this function over here within base model which adds some custom login of Business unit. So wanted to confirm if that wouldn't get affected by this migration

Comment (KevLehman) [reply apps/meteor/server/models/raw/LivechatDepartment.ts:249]: Not yet, as this only migrates the function. There's another PR for updating usages

**Review by sampaiodiego — DISMISSED** [review]

**Review by sampaiodiego — APPROVED** [review]

**Review by murtaza98 — APPROVED** [review]

**Pull request merged**